### PR TITLE
Insights report list rewritten as a table

### PR DESCRIPTION
**Pull request by AllenBW**
fixes https://projects.engineering.redhat.com/browse/RHCLOUD-1186
follows design here: https://redhat.invisionapp.com/share/V4NY3C6XKRW#/screens/356890434

we're gonna save search and filter for phase two of this work 👍 

### ~todo~ nada left to do
* ~proper expand collapse in kebab menu~
* ~card styling still has that wrenched box-shadow~
* ~empty state stuff~
* ~searching~ gonna save this for v2
* ~filtering~ gonna save this for v2 

### full screen screenshot (for those in need of)
Screen Shot 2019-05-14 at 12 15 08 PM
Screen Shot 2019-05-14 at 12 20 56 PM

### no rule hits, its a good thing
Screen Shot 2019-05-14 at 11 53 32 AM

### error fetching (absent entity looks the same, different text )
Screen Shot 2019-05-14 at 11 58 32 AM

### our expand and collapse, by default all are expanded

Screen Shot 2019-05-14 at 12 00 26 PM

### rest of the report looks the same
Screen Shot 2019-05-14 at 12 00 57 PM

### CODE REVIEW

**Comment by AllenBW** [comment]
cc @katierik  @robcwilliams

**Comment by Robcwilliams** [comment]
@AllenBW I like the new layout. A few comments

- If we can add spacing between the icons and the words, they're crunched up together at the moment 

- The ansible remediation iconography / colors confuse me. There is a green checkbox and a black 
checkbox... Not sure what the difference is, but if the black checkbox is meant to say there is no ansible support, this confuses me due to it being a check

**Comment by AllenBW** [comment]
@Robcwilliams did the spacing thing! check the second ss for updates

**Comment by katierik** [comment]
- There should be nothing when there is no Ansible support (not a black check).
- Ansible text should be the "small" pf text
- Also the labels for "impact","likelihood","total risk" and "risk of change" should be the small type
- Text should be "Ansible remediation" ( not "Ansible Remediation") - Sentence case!
- Why is "ansible remediation" wrapped some places and not others? It should ideally all respond uniformly if it needs to wrap due to screen size.
- Is "Collapse all" and "expand all" in here? Can't tell.
- In the mock there is 16px between the rule title and the impact/likelihood/total risk icons. 

Xnip2019-05-14_12-24-46

Xnip2019-05-14_12-27-02

**Comment by AllenBW** [comment]
@katierik  updated screen shot below, changed what we can in the scope of this pr, 
* fixed ansible support size/ case/visibility, thanks for the clarification
* fixed label size
* screen shot below shows the collapse all and expand all (also original post had this)
* added 16px top margin to the battery stack
Screen Shot 2019-05-14 at 1 13 03 PM

**Comment by katierik** [comment]
💯 awesome!

The labels with the icons should be middle aligned with each other (wasn't as noticeable before the text size change). Otherwise 👍

**Comment by AllenBW** [comment]
That'll have to change over https://github.com/RedHatInsights/insights-frontend-components/blob/master/doc/components/battery.md not something we can so much touch in this work...

**Comment by AllenBW** [comment]
@ryelo @karelhala whenever yah all can get a chance,  <3 🙇‍♀ 👍

File: src/SmartComponents/Inventory/applications/advisor/Advisor.js [hunk 1]
```diff
@@ -79,84 +94,150 @@ class InventoryRuleList extends Component {
...
+                    </CardBody>
+                </Card>
+            ) }
+            { inventoryReportFetchStatus === 'fulfilled' && (results > 0 ?
+                <Table aria-label={ 'rule-table' } onCollapse={ this.handleOnCollapse } rows={ rows } cells={ [ '' ] }>
+                    <TableHeader/>
+                    <TableBody/>
+                </Table>
+                :
+                <Card>
```
Comment (karelhala) [inline src/SmartComponents/Inventory/applications/advisor/Advisor.js]: Is it correct in Card? Other empty states are outside card component.

File: src/SmartComponents/Inventory/applications/advisor/Advisor.js [hunk 2]
```diff
@@ -79,84 +94,150 @@ class InventoryRuleList extends Component {
...
-            </Fragment>
-        );
+                    isOpen: true,
+                    cells: [
+                        {
+                            title: <Fragment>
+                                <Split>
+                                    <SplitItem> { rule.category.name } &gt; </SplitItem>
+                                    <SplitItem isFilled> { rule.description } </SplitItem>
+                                    <SplitItem style={ { fontSize: global_FontSize_sm.value } }>
```
[the remark this answers is not in the document]
Comment (AllenBW) [reply src/SmartComponents/Inventory/applications/advisor/Advisor.js]: down with either 🤷‍♂

File: src/SmartComponents/Inventory/applications/advisor/Advisor.js (hunk 1 above)
Comment (AllenBW) [reply src/SmartComponents/Inventory/applications/advisor/Advisor.js]: We need to have the white background and margin around the the message state, i didn't build it into messagestate.js (cuz maybe we wanta abstract it somewhere else n drop it somewhere else 🤷‍♂) by default empty states don't have a background color

File: src/SmartComponents/Inventory/applications/advisor/Advisor.js [hunk 3]
```diff
@@ -79,84 +94,150 @@ class InventoryRuleList extends Component {
...
+                                        { value.resolution.has_playbook &&
+                                        <Fragment><CheckIcon style={ { color: global_success_color_200.value } }/> Ansible remediation </Fragment>
+                                        }
+                                    </SplitItem>
+                                </Split>
+                                <Section type='icon-group__with-major' style={ { fontSize: global_FontSize_sm.value, marginTop: '16px' } }>
```
[the remark this answers is not in the document]
Comment (AllenBW) [reply src/SmartComponents/Inventory/applications/advisor/Advisor.js]: I can do this... but for whatever reason.. in dev, i don't see these styles applied when they are done via a class
Screen Shot 2019-05-15 at 7 46 19 AM

(see the highlighted row below)

[the remark this answers is not in the document]
Comment (AllenBW) [reply src/SmartComponents/Inventory/applications/advisor/Advisor.js]: but i do see them when applied directly as styles to elements

File: src/SmartComponents/Inventory/applications/advisor/Advisor.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ryelo) [reply src/SmartComponents/Inventory/applications/advisor/Advisor.js]: Stylesheets over inline styles every day of the week for me 😄

File: src/SmartComponents/Inventory/applications/advisor/Advisor.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryelo) [reply src/SmartComponents/Inventory/applications/advisor/Advisor.js]: Something might not be scoped right or the specificity isn't correct. This page is using other styles from `advisor.scss`.

File: src/SmartComponents/Inventory/applications/advisor/Advisor.js (hunk 1 above)
Comment (ryelo) [reply src/SmartComponents/Inventory/applications/advisor/Advisor.js]: Was this in the UX mocks? Other places aren't using backgrounds on their empty states. For consistency, this should be revisited.

**Comment by karelhala** [comment]
@AllenBW thanks for the screenshots! Can you remove the Remediate with ansible and kebab button if there are no rules?

File: src/SmartComponents/Inventory/applications/advisor/Advisor.js (hunk 1 above)
Comment (AllenBW) [reply src/SmartComponents/Inventory/applications/advisor/Advisor.js]: @katierik can you confirm or deny, white background for empty states?

Comment (katierik) [reply src/SmartComponents/Inventory/applications/advisor/Advisor.js]: Yes white background. [Mock](https://redhat.invisionapp.com/share/V4NY3C6XKRW#/356911950_Empty_State_-_No_Insights). 
The difference between a grey bg and a white bg is this:
- Empty state should be directly on the grey bg if there is a configuration reason it is empty (i.e. it isn't connected yet)
- Empty state should be on the white bg if it is properly configured, but there just isn't anything to show! (e.g. no results after filtering; the system/rule happens to have 0 hits)

**Comment by AllenBW** [comment]
@karelhala YOUR WISH IS MY COMMAND!! 
Screen Shot 2019-05-15 at 10 39 55 AM

**Comment by katierik** [comment]
Not a blocker I don't think, but we should still have a toolbar visible on the page even when empty. Will be more relevant when we have filter etc on there. [Mock](https://redhat.invisionapp.com/share/V4NY3C6XKRW#/356911950_Empty_State_-_No_Insights) has it. 
Its part of the empty state pattern.

**Comment by AllenBW** [comment]
@katierik and here's our failed state
Screen Shot 2019-05-15 at 10 51 52 AM

**Comment by nlcwong** [comment]
@AllenBW hey Allen, when we get the "failed state" what http error code is the UI getting?

**Comment by AllenBW** [comment]
@nlcwong anything that is not a `20x` 🤔 

@katierik with love from keyboard
Screen Shot 2019-05-15 at 12 25 13 PM

**Pull request merged**